p2plending/WeiLend.sol

```diff
@@ -0,0 +1,187 @@
+// WeiLend System
+// Start, lend, payout and return interest yield to funders
+// @creator of the original WeiFund code:
+// Nick Dodson <thenickdodson@gmail.com>
+// @forked to implement the WeiLend by:
+// M. Terzi <ma.terzi@tiscali.it> ===> http://github.com/terzim
+// If goal is not reached and campaign is expired, contributers can get their donation refunded individually
+// If goal is reached by alloted time, contributions can still be made
+// After a grace period, the beneficiary returns funds to the funders in installments at a given interest rate
+
+contract WeiLend {
+// builds a variable (struct) called User which has two sub-variables, an integer number called numLoans and a mapping of 
+    // loans.
+    
+    struct User
+    {
+        uint numLoans;
+        mapping(uint => uint) loans;
+    }
+    
+    // builds a variable (struct) called Funder which is identified by an address (his public "signature"), and an amount which he has lent
+    // which will define his/her entitlement to the installments repayments. 
+    
+    struct Funder 
+    {
+        address addr;
+        uint amount;
+    }
+    
+     // builds a variable (struct) called Loan. This is the most complex variable of the set. The most important here are the sub-variables:
+     // timelimit [the time for which the Loan is open to funding], fundingGoal [the amount that the guy who gets the
+     // loan wants to collect], and the key contractual features of the loan which are the interest_rate, the        
+     // gracePeriod and the tenor. 
+     
+     // importantly, this variable contains a map of the funders and a pointer to the funders. 
+     
+    struct Loan 
+    {
+        bytes32 operationName;
+        address beneficiary;
+        uint timelimit;
```

Comment:
If smaller types suffice for some of these members, the contract will be much cheaper.